src/main/spark2.1/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -146,7 +146,8 @@ case class FileSourceScanExec(
     outputSchema: StructType,
     partitionFilters: Seq[Expression],
     dataFilters: Seq[Filter],
-    override val metastoreTableIdentifier: Option[TableIdentifier])
+    override val metastoreTableIdentifier: Option[TableIdentifier],
+    val backPortForOrcColumnarBatch: Boolean = false)
```

Comment:
Why we need this flag? 
I saw it complex the code for handling both cases. Do we have to?

Reply:
The parquet columnar batch reader in OAP is using the columnar batch from Spark2.1. The orc columnar batch reader in OAP is using the columnar batch from Spark2.3 In FileSourceScanExec, it needs to deal with these two classes when iterating the rows.